File: apis/src/main/java/com/beat/apis/booking/application/dto/event/BookingCreatedEvent.java
package com.beat.apis.booking.application.dto.event;

import java.time.LocalDateTime;

public record BookingCreatedEvent(
	LocalDateTime bookingDateTime,
	String performanceTitle,
	int purchaseTicketCount,
	String bookerName,
	String scheduleDisplayName,
	int currentSoldTicketCount,
	int totalTicketCount
) {
	public static BookingCreatedEvent of(LocalDateTime bookingDateTime, String performanceTitle, int purchaseTicketCount,
		String bookerName, String scheduleDisplayName, int currentSoldTicketCount, int totalTicketCount) {
		return new BookingCreatedEvent(
			bookingDateTime,
			performanceTitle,
			purchaseTicketCount,
			bookerName,
			scheduleDisplayName,
			currentSoldTicketCount,
			totalTicketCount
		);
	}
}

File: application/frontoffice/src/main/kotlin/com/beat/application/frontoffice/booking/event/BookingCreatedEvent.kt
package com.beat.application.frontoffice.booking.event

import java.time.LocalDateTime

data class BookingCreatedEvent(
    val bookingDateTime: LocalDateTime,
    val performanceTitle: String,
    val purchaseTicketCount: Int,
    val bookerName: String,
    val scheduleDisplayName: String,
    val currentSoldTicketCount: Int,
    val totalTicketCount: Int,
)
